### PR TITLE
Refactor Configuration Handling

### PR DESCRIPTION
**Pull request by scanon**
This refactors the configuration handling to use a class instead of a dict.  So this is a large change because this weaves through out the platform. 

JobRunner/config.py add a Config class.  This is initialized by the CLI tools or modules.  The config object is then passed off to the other modules.  Many of the tests also needed to be updated since they were initializing a configuration dict before.
 

There are a few minor flake8 fixes too but this is primarily what is needed to make the configuration change and fix tests.

### CODE REVIEW

**Review by MrCreosote — COMMENTED** [review]
This review was more about code structure / etc vs. correctness, since I'm not at all an expert on this code base.

I went through the entire set of changes, but there's probably some stuff that I missed. This is a lot of comments already so I'll stop here

File: test/test_jobrunner.py
[the remark this answers is not in the document]
Comment (scanon) [reply test/test_jobrunner.py:1]: Yes.  I think I want to move these over to pytest.  I think I will do a separate pass on that and add some fixtures.

File: test/test_jobrunner.py [hunk 1]
```diff
@@ -98,11 +96,9 @@ def test_run_sub(self, mock_ee2, mock_auth):
         params["method"] = "RunTester.run_RunTester"
         params["params"] = [{"depth": 3, "size": 1000, "parallel": 5}]
         config = deepcopy(EE2_LIST_CONFIG)
```
[the remark this answers is not in the document]
Comment (scanon) [reply test/test_jobrunner.py:98]: I'll save that for when I port to pytest.

File: JobRunner/Callback.py [hunk 2]
```diff
@@ -1,47 +1,19 @@
 import os
```
[the remark this answers is not in the document]
Comment (scanon) [reply JobRunner/Callback.py:1]: I'll make a note to add that in a separate PR

File: JobRunner/config.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    workdir = None
+    base_url = "https://ci.kbase.us/services/"
+    ee2_url = None
+    token = None
+    admin_token = None
+    runtime = None
+    hostname = None
+    debug = False
+    cgroup = None
+    user = None
```
[the remark this answers is not in the document]
Comment (scanon) [reply JobRunner/config.py]: Removed.

**Pull request merged**